### DISCUSSION TITLE
Detect if AIR app is running in elevated mode.

### DISCUSSION THREAD

**Discussion by manishpatel0404** [airsdk/Adobe-Runtime-Support#2320] [Q&A] [answered]
Is there a method to detect if an AIR app is running in elevated mode on Windows Desktop?

**Comment by itlancer** [comment, marked as answer]
@manishpatel0404, you can write your own C++ native extension (ANE) to detect it. Working C++ code for that: https://www.codeproject.com/Answers/545266/IsplusProcessplusRunningplusasplusAdministrator#answer3
Also you can try to launch cmd via `NativeProcess` and try to get such infromation: https://stackoverflow.com/questions/7985755/how-to-detect-if-cmd-is-running-as-administrator-has-elevated-privileges

**Reply by manishpatel0404** [reply]
Thanks @itlancer 

Ended up adding a function to my ANE.

Some more resources for this subject that can be useful to others:

https://learn.microsoft.com/en-us/windows/win32/api/securitybaseapi/nf-securitybaseapi-checktokenmembership

https://stackoverflow.com/questions/992661/using-checktokenmembership-return-always-true-even-if-the-process-user-is-not-ad

https://web.archive.org/web/20100418044859/http://msdn.microsoft.com/en-us/windows/ff420334.aspx

**Comment by RayGyoe** [comment]
fyi:https://github.com/RayGyoe/ANEWinCore/blob/master/native/ANEQcloudLiveVS2010/ANEWinCore.cpp#L382

**Reply by henke37** [reply]
Note that this code specifically checks if the token is marked as elevated. This only applies when UAC is enabled.

I recommend against worrying about elevation and instead checking the actual permission you need. Usually by just doing whatever you needed the permission in the first place for.
